### DISCUSSION TITLE
My Stream Deck doesn't seem to have a USBMODEM device on my iMac

### DISCUSSION THREAD

**Discussion by aseedig** [elgatosf/elgato#12] [Help]
Hi community, 

My Stream Deck is working fine so far. However, when I use Python to connect to the device, I always get the probe error. All required plugins are installed but it seems Stream Deck is not listenig on any port. I cannot find a hint of the stream deck usbmodem on my system. The error says:

StreamDeck.DeviceManager.ProbeError: ('Probe failed to find any functional HID backend.', {'libusb': TransportError("No suitable LibUSB HIDAPI library found on this system. Is the 'libhidapi.dylib' library installed?")})

I have installed the Stream Deck software version 6.1.0 (18521). The Firmware on my Deck is 1.01.000. The iMac is on OSX Ventura 13.3.1 (22E261). I have all the libraries required installed and spent my weekend trouble shooting the issue. I came to the conclusion something is not right. Is there an additional software to be installed on the Mac to have the USB modem initialised when Stream Deck starts? As I said, all the libraries required to work under Python3 are installed and all dependncies seem to be there. But no usbmodem.

Any help is appreciated.

Kind regards

Axel

**Comment by ChekTek** [comment]
Hi Axel,

Unfortunately, we don't have any HID documentation to share publicly, however, if you are using Python, you might find this project helpful: https://github.com/abcminiuser/python-elgato-streamdeck.

Best,
Zack

**Comment by aseedig** [comment]
Hi Zack,

sorry for the late reply. I was out on a business trip and just returned. 
I have downloaded the stuff from the link. 

The streamdeck library is already installed on my system. All the depencies are fullfilled. However, when I try to run one of the example scripts (exampe_deckinfo.py) I get an error message:

```
aseedig@DESKTOP-HBIE53J src % python3 example_deckinfo.py
Found 1 Stream Deck(s).

Traceback (most recent call last):
  File "/Users/aseedig/Downloads/StreamDeck/python-elgato-streamdeck-master/src/example_deckinfo.py", line 52, in <module>
    deck.open()
  File "/Users/aseedig/Downloads/StreamDeck/python-elgato-streamdeck-master/src/StreamDeck/Devices/StreamDeck.py", line 155, in open
    self.device.open()
  File "/Users/aseedig/Downloads/StreamDeck/python-elgato-streamdeck-master/src/StreamDeck/Transport/LibUSBHIDAPI.py", line 384, in open
    self.device_handle = self.hidapi.open_device(self.device_info['path'])
  File "/Users/aseedig/Downloads/StreamDeck/python-elgato-streamdeck-master/src/StreamDeck/Transport/LibUSBHIDAPI.py", line 213, in open_device
    raise TransportError("Could not open HID device.")
StreamDeck.Transport.Transport.TransportError: Could not open HID device.
```
I am not sure what I am missing here. I wanted to use the stream deck xl with X-Plane and use the PilotDeck solution. That is the point where I get the [...]  raise ProbeError("Probe failed to find any functional HID backend.", probe_errors) [...] error.

This error was the reason I contacted Elgato Support and they pointed me to GitHub. I went to so many different troubleshooting processes so far and your attempt seems to be another dead end. 

Don't get me wrong, I am not blaming you. Quite the opposite. I am grateful for any help, but I usually program in SQL and Java script and not in Python. I do Python for some minor things and have written some Python scripts in the past, but I am far from professional in Python.

Maybe you can help me here?!

Looking forward to hear from you.

Best

Axel

**Comment by aseedig** [comment]
Hi Zack,

I did some more testing. I did quit the stream deck software and the scripts are working.
Do I assume correctly, that the Python scripts only work, if there is no Elgato Stream Deck Software running?

If so, how does that work with plugins? Is this a limitation? Is there a way to get this working together?

Thanks in advance.

Kind regards

Axel
